### PR TITLE
Cmake is the worst.

### PR DESCRIPTION
**Pull request by JonathanHenson**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -232,7 +232,7 @@ configure_file(${CONFIG_HEADER_TEMPLATE}
         ESCAPE_QUOTES)
 
 if (NOT CMAKE_CROSSCOMPILING)
-    if (BUILD_TESTING)
+    if (NOT DEFINED BUILD_TESTING OR BUILD_TESTING)
         include(CTest)
         add_subdirectory(tests)
     endif()
```
Comment (graebm) [inline CMakeLists.txt]: CTest defines the `BUILD_TESTING` option, should this be:
```
if (NOT CMAKE_CROSSCOMPILING)
    include(CTest)
    if (BUILD_TESTING)
        add_subdirectory(tests)
    endif()
endif()
```
???

Comment (JonathanHenson) [reply CMakeLists.txt]: no it shouldn't, otherwise this PR wouldn't exist

Comment (graebm) [reply CMakeLists.txt]: https://cmake.org/cmake/help/v3.1/module/CTest.html

Comment (JonathanHenson) [reply CMakeLists.txt]: I don't care what the docs say.... they;re wrong

Comment (JonathanHenson) [reply CMakeLists.txt]: we could move include(CTest) to the top of the project if you'd like

**Pull request closed without merging**